Testing/functional/tests/cypress/integration/e2e/timeline/dependent.js

```diff
@@ -0,0 +1,254 @@
+const { AuthMethod, Dataset } = require("../../../support/constants");
+
+const authorizedDependentHdid = "162346565465464564565463257";
+const unauthorizedDependentHdid = "BNV554213556";
+
+const homePath = "/home";
+const unauthorizedPath = "/unauthorized";
+
+function assertDatasetPresence(dataset, expected = true) {
+    const filterSelector = `[data-testid=${toPascalCase(dataset)}-filter]`;
+    const timelineCardSelector = `[data-testid=${dataset.toLowerCase()}Title]`;
+
+    cy.get("[data-testid=filterContainer]").should("not.exist");
+    cy.get("[data-testid=filterDropdown]").click();
+    cy.get("[data-testid=filterContainer]").should("be.visible");
+
+    if (expected) {
+        cy.get(filterSelector).should("exist").click({ force: true });
+        cy.get("[data-testid=btnFilterApply]").should("be.visible").click();
+        cy.get(timelineCardSelector).should("be.visible");
+        cy.get("[data-testid=clear-filters-button]")
+            .should("be.visible")
+            .click();
+    } else {
+        cy.get(filterSelector).should("not.exist");
+        cy.get("[data-testid=btnFilterCancel]").click();
+    }
+}
+
+function toPascalCase(s) {
+    return s.charAt(0).toUpperCase() + s.slice(1);
```

Comment:
I thought you made the util case insensitive :)

Reply:
The utility is case-insenstive; however, there's one place where they are referenced in data-testids without being lowercased.